agent/service_test.go

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Ultraviolet
+// SPDX-License-Identifier: Apache-2.0
+package agent_test
```

Comment:
you can use agent package so that you don't have to use new to initilize service or if you need to transition state artificially.